src/primitives.jl

```diff
@@ -179,6 +181,30 @@ end
 dropnull(X::NullableVector) = X.values[!X.isnull]    # -> Vector
 
 """
+    dropnull!(X::AbstractVector)
+
+Remove null entries of `X` in-place and return the updated vector. This is a
+no-op if no nulls are present.
+
+    dropnull!(X::NullableVector)
+
+Remove null entries of `X` in-place and return a `Vector` view of `X.values`
+"""
+function dropnull!(X::AbstractVector)
+    if !(Nullable <: eltype(X))
+        return X
+    end
+    deleteat!(X, find(isnull, X))
```

Comment:
For consistency, this function should also unwrap the values in the returned vector, after making a copy. This should always work since the condition at the top will only work for `Any`.